chapter2/chapter2.Rmd

```diff
@@ -327,87 +483,282 @@ par(mar = om)
+  labs(x = "Life Expectancy", y = "Frequency", title = "Histogram of life expectancy in 2015") +
+  theme_bw(base_size = 16)
+```
+]
+
+---
+
+# `ggplot2`: Fancy Histogram with `facet_grid()`
+
+.pull-left[
+```{r, echo = TRUE, eval = FALSE}
+library(ggplot2)
+
+ggplot(gapminder_2015,
+       aes(x = life_expectancy)) +
+  geom_histogram(binwidth = 5,
+                 boundary = 45,
+                 colour = "white",
+                 fill = "#d90502") +
+  labs(x = "Life Expectancy",
+       y = "Frequency",
+       title = "Histogram of life expectancy in 2015") +
+  theme_bw(base_size = 16) +
+  facet_grid(rows = vars(continent))
+```
+]
+
+.pull-right[
+```{r, echo = FALSE, eval = TRUE, fig.height = 7}
+ggplot(gapminder_2015,
+       aes(x = life_expectancy)) +
+  geom_histogram(binwidth = 5,
+                 boundary = 45,
+                 colour = "white",
+                 fill = "#d90502") +
+  labs(x = "Life Expectancy",
+       y = "Frequency",
+       title = "Histogram of life expectancy in 2015") +
+  theme_bw(base_size = 16) +
+  facet_grid(rows = vars(continent))
+```
+]
+
+---
+
+# `ggplot2`: Boxplots
+
+.pull-left[
+```{r, echo = TRUE, eval = FALSE}
+ggplot(gapminder_2015,
+       aes(x = continent, y = life_expectancy)) +
+  geom_boxplot(colour = "black",
+               fill = "#d90502") +
+  labs(x = "Continent",
+       y = "Life expectancy in 2015", 
+       title = "Life expectancy by continent in 2015") +
+  theme_bw(base_size = 20)
+```
+]
+
+.pull-right[
+```{r, echo = FALSE, eval = TRUE, fig.height = 5}
+ggplot(gapminder_2015,
+       aes(x = continent, y = life_expectancy)) +
+  geom_boxplot(colour = "black", fill = "#d90502") +
+  labs(x = "Continent", y = "Life expectancy in 2015", title = "Life expectancy by continent in 2015") +
+  theme_bw(base_size = 20)
+```
+]
+
+---
+
+# `ggplot2`: Scatter Plots
+
+
+.pull-left[
+```{r, echo = TRUE, eval = FALSE}
+ggplot(gapminder_2015,
+       aes(x = fertility,
+           y = infant_mortality)) +
+  geom_point(size = 3,
+             alpha = 0.5,
+             colour = "#d90502") +
+  labs(x = "Infant mortality",
+       y = "Fertility rate",
+       title = "Relationship between fertility and infant mortality in 2015") +
+  theme_bw(base_size = 16)
+```
+]
+
+.pull-right[
+```{r, echo = FALSE, eval = TRUE, fig.height = 5}
+ggplot(gapminder_2015,
+       aes(x = fertility, y = infant_mortality)) +
+  geom_point(size = 3, alpha = 0.5, colour = "#d90502") +
+  labs(x = "Infant mortality", y = "Fertility rate",
+       title = "Relationship between fertility and infant mortality in 2015") +
+  theme_bw(base_size = 16)
+```
+]
+
+
+---
+
 class: inverse, middle
 
 
 # It's Tutorial Time!
 
 ---
 
-# Tutorial Chapter 2
+# Tutorial 1 (10 minutes)
+
+Time for our first tutorial!!
 
-Time for our first tutorial!! Type this into your `RStudio` console:
+Type this into your `RStudio` console:
 
-```{r,eval=FALSE}
+```{r, echo = TRUE, eval=FALSE}
```

Comment:
you guys need to make sure this runs on your machines. the unit tests don't pick this up.

Reply:
I don't understand what you mean?
I've run the tutorial and it works on my end. There's only one thing which doesn't, in the "Continuous Data" tab, section "The Standard Deviation is measured in same units as the Mean" where `var(euros)` and `fex^2` return an error because neither objects are found.
Also, I think there's a small typo in the "Discrete Data" tab, "Let’s denote p_k the proportion of all entries in x that have value k. i.e. p_1 = number of times x = ***1*** (not x_1) / N.

Reply:
yes that's what i meant!